lib/index.js

```diff
@@ -74,33 +71,39 @@ const Popover = createReactClass({
     style: T.object,
     tipSize: T.number,
     onOuterAction: T.func,
-  },
-  mixins: [ReactLayerMixin()],
-  getDefaultProps () {
-    return {
-      tipSize: 7,
-      preferPlace: null,
-      place: null,
-      offset: 4,
-      isOpen: false,
-      onOuterAction: function noOperation () {},
-      enterExitTransitionDurationMs: 500,
-      children: null,
-      refreshIntervalMs: 200,
-    }
-  },
-  getInitialState () {
-    return {
+  }
+  static defaultProps = {
+    tipSize: 7,
+    preferPlace: null,
+    place: null,
+    offset: 4,
+    isOpen: false,
+    onOuterAction: Utils.noop,
+    enterExitTransitionDurationMs: 500,
+    children: null,
+    refreshIntervalMs: 200,
+    appendTarget: isServer ? null : document.body
+  }
+  constructor (props) {
+    super(props)
+    this.checkTargetReposition = this.checkTargetReposition.bind(this)
```

Comment:
if you use arrow functions then you wouldn't need to bind functions in the constructor:

```js
checkTargetReposition = () => {
  // ...
}
```